### ISSUE TITLE
Device Back button not working on Android

### ISSUE THREAD

**Issue by subsub** [jitsi/jitsi-meet#1892]
The device back button is not working properly. The instance `reactInstanceManager` is not null and the `onBackPressed` method returns true from JitsiMeetView instance, but the app doesn't go to the previous screen when I'm on Conference screen. Even when inside welcome screen, pressing back button doesn't exit the app.

**Comment by saghul** [comment]
This is a known issue. 👍

**Comment by subsub** [comment]
@saghul following up my question.
Since back button is not working, How do I invoke to leave conference from native code (java) to react code?

I do the following:
Inside module `org.jitsi.meet.sdk.externalapi.ExternalAPIModule` I created this `ReactMethod` which is looked like this:
```java
public class ExternalAPIModule extends ReactContextBaseJavaModule {
    ...
    private static Callback onBackCallback;

    @ReactMethod
    public void backButtonCallback(Callback event) {
        onBackCallback = event;
    }

    public static Callback getOnBackCallback() {
        return onBackCallback;
    }
}
```
When user presses back button, it calls that callback:
```java
public class JitsiMeetActivity extends AppCompatActivity {
    public void onBackPressed() {
            ExternalAPIModule.getOnBackCallback().invoke(true);
    }
}
```
And then inside `react/features/mobile/external-api/middleware.js` I add this method:
```javascript
...
switch (action.type) {
    case CONFERENCE_FAILED:
    case CONFERENCE_JOINED:
    case CONFERENCE_LEFT:
    case CONFERENCE_WILL_JOIN:
        // Register Callback from native android
        // so that when user click on back button this will be fired
        console.warn("Conference will join. Register back button callback");
        _backButtonCallback();
    case CONFERENCE_WILL_LEAVE:
}
...
function _backButtonCallback() {
    console.warn("inside backbuttoncallback");
    NativeModules.ExternalAPI.backButtonCallback( back => {
        console.warn("Back Button Callback invoked.");
        if (back) {
            // TODO: INVOKE CONFERENCE_LEFT HERE
            console.warn("User pressed back button. What should I do? ");
        }
    });
}
```
The callback works, But I don't know what to do next in order to leave the conference.
The idea is to invoke `connection.disconnect` method or something equal with what happens when user presses 'hangup' button. But I don't know how to call method of hangup button.
Redux gets me confused so very much.
Am I in the right track? (Worth to mention as well: I'm new to React :))
Thank you

**Comment by lyubomir** [comment]
As of https://github.com/jitsi/jitsi-meet/commit/03d337612bb63003184630e374b294d7a7551765 which is shipped in Jitsi Meet for Android 1.10, the Back button is handled in the app by cancelling `Dialogs`, leaving the conference, and finishing the associated `Activity` when on the `WelcomePage`. The last one may be overridden by clients/consumers of Jitsi Meet SDK for Android by passing their own implementation of `DefaultHardwareBackBtnHandler` to `onHostResume` at https://github.com/jitsi/jitsi-meet/blob/master/android/sdk/src/main/java/org/jitsi/meet/sdk/JitsiMeetView.java#L202.
